### PR TITLE
update for documentation

### PR DESCRIPTION
**Pull request by switowski**
Draft of the documentation that I'm still refining.

### CODE REVIEW

File: invenio_search/__init__.py [hunk 1]
```diff
@@ -22,7 +22,257 @@
...
+        if 'q' in request.values:
+            search = search.query(QueryString(query=request.values.get('q')))
+
+        return jsonify(search.execute().to_dict())
+
+Run example development server:
+
+.. code-block:: console
+
+   $ FLASK_APP=app.py flask run --debugger -p 5000
```
Comment (lnielsen) [inline invenio_search/__init__.py]: ``FLASK_DEBUG=1 FLASK_APP=app.py flask run`` is the "new" way of doing it. Not sure ``--debugger`` works anymore.

**Comment by switowski** [comment]
It's no longer WIP

File: invenio_search/__init__.py [hunk 2]
```diff
@@ -22,7 +22,257 @@
...
+This will create an empty index, which is not very useful, so let's add some
+mappings. To not reinvent the wheel, let's reuse the mappings from the example
+application. Add the following line at the end of ``app.py`` file that you
+have just created:
+
+.. code-block:: python
+
+    # app.py
+    ...
+    search.register_mappings('demo', 'examples.data.elastic_v2')
```
Comment (lnielsen) [inline invenio_search/__init__.py]: It's not so clear where the mapping files coming from. Perhaps just mentioning something like.... "Above looks into directory x,y, z and disocvers files a,b, cwhich are elasticsearch mappings"

**Comment by switowski** [comment]
I have updated the docs based on the comments from @lnielsen .

File: invenio_search/__init__.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2015, 2016, 2017 CERN.
```
Comment (drjova) [inline invenio_search/__init__.py]: you only need to add the year you made the change :P

Comment (switowski) [reply invenio_search/__init__.py]: Ahh, I didn't know that! I thought we always add all the missing years. Thanks!

Comment (drjova) [reply invenio_search/__init__.py]: it's just a detail :)

Comment (switowski) [reply invenio_search/__init__.py]: Anyway, this file was also touched in `2016` :) (23acaa61301276a325e90)

File: invenio_search/__init__.py [hunk 4]
```diff
@@ -22,7 +22,263 @@
...
+    @app.route('/', methods=['GET', 'POST'])
+    def index():
+        search = PublicSearch()
+        ...
+
+Now, open the browser and search for documents with ``test`` in the body.
+
+.. code-block:: console
+
+   $ open http://localhost:5000/?q=body:test
```
Comment (drjova) [inline invenio_search/__init__.py]: doesn't this alias work everywhere? Shall we add something more generic like ``curl``? :)

Comment (switowski) [reply invenio_search/__init__.py]: That's a good point and I think the problem is bigger than this one example:
```console
$ ack "open http" .
new-cookiecutter/invenio-search/examples/app.py
50:    $ open http://127.0.0.1:5000/

invenio-admin/examples/app.py
50:    $ open http://127.0.0.1:5000/

elasticsearch/elasticsearch-head/README.textile
21:* @open http://localhost:9200/_plugin/head/@

invenio-oauth2server/examples/consumer.py
44:   $ open http://127.0.0.1:5000/account/settings/applications
79:    $ open http://127.0.0.1:5000/logout

invenio-oauth2server/examples/app.py
44:   $ open http://127.0.0.1:5000/account/settings/applications

invenio-search/examples/app.py
51:   $ open http://localhost:5000/?q=body:test

invenio-search/invenio_search/__init__.py
202:   $ open http://localhost:5000/?q=body:test
258:   $ open http://localhost:5000/?q=body:test

invenio-accounts/examples/app.py
55:    $ open http://127.0.0.1:5000/

invenio-access/examples/app.py
54:    $ open http://127.0.0.1:5000/

invenio-userprofiles/examples/app.py
54:    $ open http://127.0.0.1:5000/
63:    $ open http://127.0.0.1:5000/admin

invenio-pidstore/examples/app.py
51:    $ open http://0.0.0.0:5000/admin/recordmetadata/

cookiecutter-invenio-module/{{ cookiecutter.project_shortname }}/examples/app.py
27:    $ open http://127.0.0.1:5000/

```
I think we should decide what to do with this and change it in the `cookiecutter` (and other modules). I'm fine with curl.

Comment (akuzucan) [reply invenio_search/__init__.py]: I don't have much trouble to understand `open`. Besides it's pro-macos :)

Comment (akuzucan) [reply invenio_search/__init__.py]: Lol, that's my wifes account (im jbenito3 from my stupid phone)

Comment (switowski) [reply invenio_search/__init__.py]: Hahahahahahahahahahahaha :laughing: 
That really made my evening :D
I was wondering why some random people are commenting on documentation of `invenio-search` module :D

Comment (switowski) [reply invenio_search/__init__.py]: I have replaced `open` with `curl`

Comment (jbenito3) [reply invenio_search/__init__.py]: so I see that you didn't care about my wife's opinion :)

Comment (switowski) [reply invenio_search/__init__.py]: The fact that we use `curl` more often than `open` was the deciding factor :) Also @drjova doesn't have the same aliases as you, as he can't use `open` on his Macbook. Probably it's a part of Apple's strategy to remove useful things from Macbooks with every new generation ;)

**Pull request merged**